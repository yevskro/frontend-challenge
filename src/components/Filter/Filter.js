import React from 'react';
import styled from 'styled-components';

function Filter(props){
  return <StyledFilter role='listitem button'>
    <Text>{props.value}</Text>
    {props.remove ? <Remove><RemoveIcon/></Remove> : ''}
  </StyledFilter>
}

const StyledFilter = styled.li`
  height: 32px;
  display: flex;
  justify-content: center;
  align-items: center;
  background-color: #EAF3F2;
  margin: 10px 8px 10px 8px;
  border-radius: 5px;
`;

const Text = styled.span`
  padding: 0px 9px 0px 8px;
  color: #5EAAA6;
`;

const Remove = styled.div`
  width: 32px;
  height: 32px;
  background-color: #5DA5A4;
  display: flex;
  justify-content: center;
  align-items: center;
  border-radius: 0px 5px 5px 0px;
`

const RemoveIcon = styled.img`
  content: url('/assets/images/icon-remove.svg');
`;

export default Filter;