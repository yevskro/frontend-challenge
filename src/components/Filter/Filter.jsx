import React from 'react';
import styled from 'styled-components';

function Filter({ children, remove }) {
  return (
    <StyledFilter role='listitem button'>
      <Text>{children}</Text>
      {remove ? (
        <Remove>
          <RemoveIcon />
        </Remove>
      ) : (
        ''
      )}
    </StyledFilter>
  );
}

const StyledFilter = styled.li`
  height: 32px;
  display: flex;
  justify-content: center;
  align-items: center;
  background-color: #eaf3f2;
  margin: 10px 16px 10px 0px;
  border-radius: 5px;
`;

const Text = styled.span`
  padding: 0px 9px 0px 8px;
  color: #5eaaa6;
`;

const Remove = styled.div`
  width: 32px;
  height: 32px;
  background-color: #5da5a4;
  display: flex;
  justify-content: center;
  align-items: center;
  border-radius: 0px 5px 5px 0px;
`;

const RemoveIcon = styled.img`
  content: url('/assets/images/icon-remove.svg');
`;

export default Filter;
