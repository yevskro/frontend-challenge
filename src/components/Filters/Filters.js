import React from 'react';
import styled from 'styled-components';

function Filters(props){
    return <StyledFilters>
        {props.children}
    </StyledFilters>
}

const StyledFilters = styled.div`
    display: flex;
    flex-wrap: wrap;
    width: 100%;
`;

export default Filters;