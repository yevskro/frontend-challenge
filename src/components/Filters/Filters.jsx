import React from 'react';
import styled from 'styled-components';

function Filters({ children, flexEnd }) {
  return (
    <StyledFilters flexEnd={flexEnd} role="listbox">
      {children}
    </StyledFilters>
  );
}

const StyledFilters = styled.ul`
  display: flex;
  flex-wrap: wrap;
  width: 100%;
  justify-content: ${(props) => (props.flexEnd ? 'flex-end' : 'flex-start')};
  @media (max-width: 415px) {
    justify-content: flex-start;
  }
`;

export default Filters;
