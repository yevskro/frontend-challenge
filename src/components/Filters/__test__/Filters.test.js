import React from 'react';
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer'; 
import Filters from '../Filters';
import Filter from '../../Filter/Filter';

test('renders without crashing', () => {
  const { getByRole } = render(<Filters/>);
  const filtersElement = getByRole('listbox');
  expect(filtersElement).toBeInTheDocument();
});

test('renders children', () => {
    const { getByRole } = render(<Filters><Filter value='CSS'/></Filters>);
    const filterElement = getByRole('listitem');
    expect(filterElement).toBeInTheDocument();
});

test('snapshot', async () => {
  const tree = renderer
    .create(<Filters/>)
    .toJSON();
  expect(tree).toMatchSnapshot();
})