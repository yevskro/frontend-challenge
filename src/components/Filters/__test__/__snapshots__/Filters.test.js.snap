// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  width: 100%;
}

<div
  className="c0"
  role="listbox"
/>
`;
